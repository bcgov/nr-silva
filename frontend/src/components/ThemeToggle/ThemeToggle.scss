@use '@bcgov-nr/nr-fsa-theme/design-tokens/colors.scss' as colors;
.theme-toggle {
    width: 62px;
    height: 31px;
    border-radius: 15px;
    display: flex;
    align-items: center;
    padding: 2px;
    cursor: pointer;
    position: relative;
    overflow: hidden;
  
    &.off {
      background-color: white;
    }
  
    &.on {
      background-color: black;
    }
  
    .circle {
      width: 26px;
      height: 26px;
      border-radius: 50%;
      display: flex;
      justify-content: center;
      align-items: center;
      transition: transform 0.3s, background-color 0.3s;
      background-color: #0073E6; // Default background color
    }
  
    &.off {
      .circle {
        transform: translateX(0);
      }
    }
  
    &.on {
      .circle {
        transform: translateX(31px);
        background-color: white;
      }
    }
  
    .icon {
      width: 24px;
      height: 24px;
    }
    &.on {
        .icon {
            fill: black;
        }
    }
    &.off {
        .icon {
            fill: white;
        }
    }
  }
  