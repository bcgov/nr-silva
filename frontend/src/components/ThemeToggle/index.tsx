import React, { useEffect, useState } from 'react';
import './ThemeToggle.scss';
import { AsleepFilled, LightFilled } from '@carbon/icons-react';
import { useThemePreference } from '../../utils/ThemePreference';
import { toggleTheme } from '../../utils/ThemeFunction';

const ThemeToggle = () => {
  const { theme, setTheme } = useThemePreference();
  console.log("theme:"+theme)
  const [isToggled, setIsToggled] = useState(theme === 'g10'?false:true);

  useEffect(()=>{
    setIsToggled(theme === 'g10'?false:true)
  },[])

  const handleToggle = async () => {
    toggleTheme(theme,setTheme)
    // keep the logic opposite at the time of sending the toggle
    setIsToggled(theme === 'g10'?true:false);
  };

  return (
    <div className={`theme-toggle ${isToggled ? 'on' : 'off'}`} onClick={handleToggle}>
      <div className="circle">
        {isToggled ? <AsleepFilled className="icon" /> : <LightFilled className="icon" />}
      </div>
    </div>
  );
};

export default ThemeToggle;
