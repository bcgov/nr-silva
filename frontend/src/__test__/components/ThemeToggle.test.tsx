import React from 'react';
import { render } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';
import ThemeToggle from '../../components/ThemeToggle';

describe('Theme toggle component tests', () => {
  it('should render correctly', () => {
    const { container } = render(<ThemeToggle />);
    expect(container).toMatchSnapshot();
  });

  it('should toggle theme when button is clicked', () => {
    const { container } = render(<ThemeToggle />);
    const button = container.querySelector('button');
    button?.click();
    expect(container).toMatchSnapshot();
  });
});
