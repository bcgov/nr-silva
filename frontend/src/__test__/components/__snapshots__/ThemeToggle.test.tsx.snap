// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Theme toggle component tests > should render correctly 1`] = `
<div>
  <div
    class="theme-toggle on"
    role="button"
    tabindex="0"
  >
    <div
      class="circle"
    >
      <svg
        aria-hidden="true"
        class="icon"
        fill="currentColor"
        focusable="false"
        height="16"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 32 32"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14.98,3a1.0024,1.0024,0,0,0-.1746.0156A13.0959,13.0959,0,0,0,16.63,28.9973c.1641.006.3282,0,.4909,0a13.0724,13.0724,0,0,0,10.702-5.5556,1.0094,1.0094,0,0,0-.7833-1.5644A13.08,13.08,0,0,1,15.8892,4.38,1.0149,1.0149,0,0,0,14.98,3Z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Theme toggle component tests > should toggle theme when button is clicked 1`] = `
<div>
  <div
    class="theme-toggle on"
    role="button"
    tabindex="0"
  >
    <div
      class="circle"
    >
      <svg
        aria-hidden="true"
        class="icon"
        fill="currentColor"
        focusable="false"
        height="16"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 32 32"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14.98,3a1.0024,1.0024,0,0,0-.1746.0156A13.0959,13.0959,0,0,0,16.63,28.9973c.1641.006.3282,0,.4909,0a13.0724,13.0724,0,0,0,10.702-5.5556,1.0094,1.0094,0,0,0-.7833-1.5644A13.08,13.08,0,0,1,15.8892,4.38,1.0149,1.0149,0,0,0,14.98,3Z"
        />
      </svg>
    </div>
  </div>
</div>
`;
